src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -122,22 +127,51 @@ public ResultResource(UserService userService,
             //In the future we also might want to save these results in the database
             return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
         }
+
+        Optional<Participation> participation = getParticipation(planKey);
+        if (participation.isPresent()) {
+            resultService.onResultNotified(participation.get());
+            return ResponseEntity.ok().build();
+        } else {
+            return notFound();
+        }
+    }
+
+    @PostMapping(value = "/resultsnew")
+    @Transactional
+    public ResponseEntity<?> notifyResultNew(@RequestBody Object requestBody) {
```

Comment:
@Timed is missing.
We should also think about adding a secret to Bamboo, because this call is not protected at all (you even don't need a valid plan key to execute it)

What do you think, would this be possible?

Reply:
I've added a secret that must match a bamboo-variable (see https://github.com/ls1intum/bamboo-server-notification-plugin).